### PR TITLE
Rip 118 contact telephone number

### PR DESCRIPTION
**Pull request by reggieb**
See https://eaflood.atlassian.net/browse/RIP-118

Uses [phonelib](https://github.com/daddyz/phonelib) to do the hard lifting.

### CODE REVIEW

File: app/views/flood_risk_engine/enrollments/steps/_correspondence_contact_telephone.html.erb [hunk 1]
```diff
@@ -1 +1,13 @@
-main_contact_telephone
+<fieldset>
+  <legend class="visuallyhidden"><%= t(".legend") %></legend>
+
+  <!-- Telephone number -->
+  <div class="form-group">
```
Comment (autotelik) [inline app/views/flood_risk_engine/enrollments/steps/_correspondence_contact_telephone.html.erb]: missing helper to process errors and create anchor link

File: app/forms/flood_risk_engine/steps/correspondence_contact_telephone_form.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+      property :telephone_number
+
+      # phone validation from https://github.com/daddyz/phonelib
+      validates(
+        :telephone_number,
+        phone: {
+          message: I18n.t("#{locale_key}.errors.telephone_number.invalid"),
+          allow_blank: true
+        },
```
Comment (autotelik) [inline app/forms/flood_risk_engine/steps/correspondence_contact_telephone_form.rb]: each to their own, but for me this format makes it hard to differentiate the different validators

Comment (woodpigeon) [inline app/forms/flood_risk_engine/steps/correspondence_contact_telephone_form.rb]: Happy to 🚢 once Tom's comment about missing helper addressed

File: app/views/flood_risk_engine/enrollments/steps/_correspondence_contact_telephone.html.erb (hunk 1 above)
Comment (reggieb) [inline app/views/flood_risk_engine/enrollments/steps/_correspondence_contact_telephone.html.erb]: Good point - well presented. I'll sort that on Monday.

**Comment by reggieb** [comment]
Rebased

File: app/views/flood_risk_engine/enrollments/steps/_correspondence_contact_telephone.html.erb (hunk 1 above)
Comment (reggieb) [inline app/views/flood_risk_engine/enrollments/steps/_correspondence_contact_telephone.html.erb]: I've now added the helper.

**Pull request merged**